### PR TITLE
[fix] load data through regular json request

### PR DESCRIPTION
**Pull request by dummdidumm**
using devalue's stringify/parse methods
Part of #7125 and #6477

Not sure exactly what's left for the above issues to get closed besides allowing setting cache headers again.

If we change to this we could use the same methods for interacting with form actions, allowing them to also return `Date` objects for example.

Marked as draft, @Rich-Harris feel free to take over/close.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by Rich-Harris** [comment]
> If we change to this we could use the same methods for interacting with form actions, allowing them to also return `Date` objects for example.

The wrinkle here is that if someone posts to an action manually, they'll get devalue-encoded data in return. We could handle it in SSR and `use:enhance` but would presumably need to expose something like

```js
import { decode } from '$app/forms';
```

or, more directly,

```js
import { decode } from 'devalue'; // doesn't currently exist, but decode(obj) would === parse(JSON.stringify(obj))
```

I think it's fine to merge this as-is then handle headers in a separate PR, so I'm marking this ready

File: packages/kit/src/runtime/client/client.js [hunk 1]
```diff
@@ -10,6 +16,7 @@ import { HttpError, Redirect } from '../control.js';
 import { stores } from './singletons.js';
 import { DATA_SUFFIX } from '../../constants.js';
 import { unwrap_promises } from '../../utils/promises.js';
+import * as devalue from 'devalue';
```
Comment (dummdidumm) [inline packages/kit/src/runtime/client/client.js:19]: will tree shaking remove the other stuff correctly if we write the import like this?

**Review by dummdidumm — COMMENTED** [review]
Can't approve because it's my own PR, but consider this an approval in case the answer to my question in the code-comment is "yes".
Edit: Oh and this needs a changeset.

File: packages/kit/src/runtime/client/client.js (hunk 1 above)
Comment (Rich-Harris) [reply packages/kit/src/runtime/client/client.js:19]: Yep, Rollup doesn't care either way

**Review by Rich-Harris — APPROVED** [review]

**Comment by 0gust1** [comment]
Thanks, I'll check if it fixes #7203 !

**Comment by Tal500** [comment]
I'm glad to see this PR fixes a theoretical consideration on a closed issue of mine:
fixes #6466

**Pull request merged**